pkg/ottl/ottlfuncs/func_replace_pattern_test.go

```diff
@@ -52,12 +52,27 @@ func Test_replacePattern(t *testing.T) {
 					return "passwd=*** ", nil
 				},
 			},
-			replacementPrefix: ottl.NewTestingOptional[string]("passwd="),
+			replacementFormat: ottl.NewTestingOptional[string]("passwd=%s"),
 			function:          optionalArg,
 			want: func(expectedValue pcommon.Value) {
 				expectedValue.SetStr("application passwd=0f2407f2d83337b1f757eb1754a7643ce0e8fba620bc605c54566cd6dfd838be otherarg=notsensitive key1 key2")
 			},
 		},
+		{
+			name:    "replace regex match (with hash function)",
+			target:  target,
+			pattern: `passwd\=[^\s]*`,
+			replacement: ottl.StandardStringGetter[pcommon.Value]{
+				Getter: func(context.Context, pcommon.Value) (interface{}, error) {
+					return "passwd=*** ", nil
+				},
+			},
+			replacementFormat: ottl.NewTestingOptional[string]("%s (passwd)"),
+			function:          optionalArg,
+			want: func(expectedValue pcommon.Value) {
+				expectedValue.SetStr("application 0f2407f2d83337b1f757eb1754a7643ce0e8fba620bc605c54566cd6dfd838be (passwd) otherarg=notsensitive key1 key2")
```

Comment:
support for suffixes

Reply:
@TylerHelmuth @evan-bradley  we're able to support a suffix as well using the suggested format string approach as mentioned in this [comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27820#issuecomment-1769375303)